File: backend/lib/hotelflux/domain/reserva_servicio.ex
defmodule HotelFlux.Domain.ReservaServicio do
  @moduledoc """
  Entidad de dominio INMUTABLE — Servicio asociado a una reserva.
  """

  defstruct [
    :id,
    :reserva_id,
    :producto_id,
    :dia_numero,
    :precio_unitario,
    :total,
    :fecha_servicio,
    :eliminado_en,
    :inserted_at,
    :updated_at,
    cantidad: 1,
    es_adicional: false,
    estado: "pendiente",
    eliminado: false
  ]

  def calcular_total(precio_unitario, cantidad) do
    Decimal.mult(precio_unitario, cantidad)
  end
end

File: frontend-personal/src/domain/entidades/reserva-servicio.ts
export interface ServicioReservaItem {
  id: string;
  reserva_id: string;
  producto_id: string;
  producto_nombre: string | null;
  categoria: string | null;
  dia_numero: number;
  cantidad: number;
  precio_unitario: string;
  total: string;
  es_adicional: boolean;
  estado: string;
  fecha_servicio: string;
  inserted_at: string;
}

export interface ServicioPorDia {
  dia: number;
  servicios: ServicioReservaItem[];
}
